lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -64,6 +69,7 @@ internal sealed class ContextRuntime : IDisposable
                 Optional<ContextRuntime> parentContext)
         {
             _serviceInjector = serviceInjector;
+            _injectedServices = serviceInjector.GetNamedInstance<ServicesSet, ISet<object>>();
```

Comment:
I don't get why we need bind volatile childContextConfiguration. Do you mean we should not make ContextRuntime injectable? 

Let me put my question this way:

Are we going to make ContextRuntime injectable in future? 
If yes, then is the service injector already contain the ServicesSet configuration? From current code, I think it is, right? 
Then if the ContextRuntime is injectable, we can just put ServicesSet in the constructor, You will then get the serviceSet for free, right? 

Before making ContextRuntime injectable, yes, you can use GetNamedInstance to get ServiceSet.